core/src/Cuda/Kokkos_Cuda_KernelLaunch.hpp

```diff
@@ -454,6 +455,34 @@ struct CudaParallelLaunchKernelInvoker<
 //==============================================================================
 // <editor-fold desc="CudaParallelLaunchImpl"> {{{1
 
+template <class Policy>
+std::enable_if_t<contains_desired_occupancy<typename Policy::traits>{}>
+modify_launch_configuration_if_desired_occupancy_is_specified(
+    Policy const& policy, cudaDeviceProp const& properties,
+    cudaFuncAttributes const& attributes, dim3 const& block, int& shmem,
+    bool& prefer_shmem) {
+  int const block_size        = block.x * block.y * block.z;
+  int const desired_occupancy = policy.experimental_get_desired_occupancy();
```

Comment:
@aprokop just reported this needs to be an explicit cast
```suggestion
  auto const desired_occupancy = static_cast<int>(policy.experimental_get_desired_occupancy());
```

Reply:
Or I could make the conversion implicit again.  @crtrott any preference?